File: app/src/main/java/com/nuhkoca/myapplication/di/module/ActivityBuilder.java
package com.nuhkoca.myapplication.di.module;

import com.nuhkoca.myapplication.di.scope.PerActivity;
import com.nuhkoca.myapplication.ui.main.MainActivity;
import com.nuhkoca.myapplication.ui.video.VideoActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * A {@link Module} that injects activities
 *
 * @author nuhkoca
 */
@Module
public abstract class ActivityBuilder {

    /**
     * Injects {@link MainActivity}
     *
     * @return an instance of {@link MainActivity}
     */
    @PerActivity
    @ContributesAndroidInjector
    abstract MainActivity contributesMainActivityInjector();

    /**
     * Injects {@link VideoActivity}
     *
     * @return an instance of {@link VideoActivity}
     */
    @PerActivity
    @ContributesAndroidInjector
    abstract VideoActivity contributesVideoActivityInjector();
}

File: app/src/main/java/com/nuhkoca/myapplication/di/module/ActivityBuilder.kt
package com.nuhkoca.myapplication.di.module

import com.nuhkoca.myapplication.di.scope.PerActivity
import com.nuhkoca.myapplication.ui.main.MainActivity
import com.nuhkoca.myapplication.ui.video.VideoActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * A [Module] that injects activities
 *
 * @author nuhkoca
 */
@Module
abstract class ActivityBuilder {

    /**
     * Injects [MainActivity]
     *
     * @return an instance of [MainActivity]
     */
    @PerActivity
    @ContributesAndroidInjector
    internal abstract fun contributesMainActivityInjector(): MainActivity

    /**
     * Injects [VideoActivity]
     *
     * @return an instance of [VideoActivity]
     */
    @PerActivity
    @ContributesAndroidInjector
    internal abstract fun contributesVideoActivityInjector(): VideoActivity
}
